### PR TITLE
Narayana specific checkstyle rules based on https://github.com/wildfl…

### PR DESCRIPTION
**Pull request by mmusgrov**
…y/wildfly-checkstyle-config

### CODE REVIEW

File: src/main/resources/narayana-checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+    "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
+    "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">
+
+<module name="Checker">
+
+    <property name="cacheFile" value=""/>
```
Comment (ochaloup) [inline src/main/resources/narayana-checkstyle/checkstyle.xml]: I think we could try to define the `cacheFile` as `target/checkstyle.cache`  to not use the property but still considering that. Or maybe if it's not a good idea from your perspective to have it here then I think let's completely remove this configuration from here (by default it's `null`). We can define that in the `pom.xml` in Narayana project then with property `checkstyle.cache.file`.

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  ~ 
+  ~ SPDX-License-Identifier: LGPL-2.1-only
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>io.narayana.checkstyle</groupId>
```
Comment (ochaloup) [inline pom.xml]: I would suggest to use only `io.narayana`, the fact it's a checkstyle artifact is defined in `artifactId`

File: src/main/resources/narayana-checkstyle/checkstyle.xml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        <module name="SuppressWarningsHolder"/>
+
+        <!-- Disallow @SuppressWarnings("all") -->
+        <module name="SuppressWarnings">
+            <property name="format" value="^all$"/>
+        </module>
+    </module>
+
+    <module name="SuppressionFilter">
+        <property name="file" value="/wildfly-checkstyle/suppressions.xml" />
```
Comment (ochaloup) [inline src/main/resources/narayana-checkstyle/checkstyle.xml]: this was wrong in my PR probably, the path should be `/narayana-checkstyle/suppressions.xml`

**Review by ochaloup — CHANGES_REQUESTED** [review]
I'm asking mainly for a change of my wrongly put suppression xml path.
The other two notes are my preference but they're not necessary.

**Review by jmfinelli — APPROVED** [review]

**Review by ochaloup — APPROVED** [review]

**Comment by ochaloup** [comment]
I hope we can merge this.

**Pull request merged**